types/mongorito/index.d.ts

```diff
@@ -0,0 +1,293 @@
+
+    static include(field: string | string[], value?: number): Query;
+    static include(query: object): Query; // TODO best parameter type
+
+    static exclude(field: string | string[], value?: number): Query;
+    static exclude(query: object): Query; // TODO best parameter type
+
+    static remove(query?: object): Promise<object>;
+
+    Model: ModelClass;
+    query: Array<[string, object]>;
+}
+
+// "extends Query": not actually inheritance, but more easy to implement
+export class Model extends Query {
+    /**
+     * @see Model#database
+     * @see Database#connection()
+     */
+    static getConnection(): Promise<Db>;
+
+    /**
+     * @see Db#collection(string)
+     */
+    static getCollection(): Promise<Collection<any>>;
+
+    static use(plugins?: Plugin | Plugin[]): void;
+
+    static modifyReducer(reducerModifier: ReducerModifier): void;
+
+    static query(method: string, query: Array<[string, any]>): Promise<object[]>;
+
+    /**
+     * @see mongodb.Collection#listIndexes()
+     * @see mongodb.CommandCursor#toArray()
+     */
+    static listIndexes(options?: { batchSize?: number, readPreference?: ReadPreference | string }): Promise<any[]>;
+
+    /**
+     * @see mongodb.Collection#createIndex()
+     */
+    static createIndex(fieldOrSpec: any, options?: IndexOptions): Promise<string>;
+
+    /**
+     * @see mongodb.Collection#dropIndex()
+     */
+    static dropIndex(indexName: string, options?: CommonOptions): Promise<object>;
+
+    static embeds(key: string, model: ModelClass): void;
+
+    constructor(fields?: object);
+
+    collection(): string;
+
+    getConnection(): Promise<Db>;
+
+    getCollection(): Promise<Collection<any>>;
+
+    set(key: string, value: any): void;
+    set(value: object): void;
+
+    get(key?: string): any;
+
+    unset(keys?: string | string[]): void;
+
+    toJSON(): object;
+
+    isSaved(): boolean;
+
+    save(): Promise<CreatedAction | UpdatedAction>;
+
+    increment(key: string, value?: number): Promise<RefreshedAction>;
+    increment(keys: { [key: string]: number }): Promise<RefreshedAction>;
+
+    refresh(): Promise<RefreshedAction>;
+
+    remove(): Promise<RemovedAction>;
+}
+
+export enum ActionTypes {
+    GET = "@@mongorito/GET",
+    SET = "@@mongorito/SET",
+    UNSET = "@@mongorito/UNSET",
+    REFRESH = "@@mongorito/REFRESH",
+    REFRESHED = "@@mongorito/REFRESHED",
+    SAVE = "@@mongorito/SAVE",
+    CREATE = "@@mongorito/CREATE",
+    CREATED = "@@mongorito/CREATED",
+    UPDATE = "@@mongorito/UPDATE",
+    UPDATED = "@@mongorito/UPDATED",
+    REMOVE = "@@mongorito/REMOVE",
+    REMOVED = "@@mongorito/REMOVED",
+    INCREMENT = "@@mongorito/INCREMENT",
+    CREATE_INDEX = "@@mongorito/CREATE_INDEX",
+    DROP_INDEX = "@@mongorito/DROP_INDEX",
+    LIST_INDEXES = "@@mongorito/LIST_INDEXES",
+    QUERY = "@@mongorito/QUERY",
+    CALL = "@@mongorito/CALL"
+}
+
+export interface GetAction {
+    type: string; // ActionTypes.GET
```

Comment:
You can actually make this `type: ActionTypes.GET` and enforce that `@@mongorito/GET` is the only allowable value.

Reply:
Very nice remark! (I previously searched... but not found this simple solution)
The PR is now _up to date_